core/src/main/java/org/apache/accumulo/core/clientImpl/TabletLocator.java

```diff
@@ -132,22 +137,35 @@ static synchronized void enable() {
     enabled = true;
   }
 
+  private static TabletLocator newTabletLocator(ClientContext context, TableId tableId,
+      TabletLocator parent) {
+
+    boolean useConcurrentLocator =
```

Comment:
Tried creating an experimental client property to enable/disable this new cache.  One problem with it is it makes it hard to turn this feature on server side, like to have the manager process use this cache for metadata table scans.